very basic in-browser game viewer

amnonbc opened the pull request.

amnonbc committed:
very basic in-browser game viewer

fliptopbox commented on board.html:
```diff
@@ -0,0 +1,59 @@
+    async function displayGame() {
+        // Use a very restricted Ably API Key which only allows the user to subscribe to messages on the game1
+        // channel in the chessdemo app.
+        const ably_key = '6Z_qIg.wS-GtA:DTq-7TiASaOOPat9IJ7zVm-jmuohkXW8ujbGs1zPthU';
```
Not elegant, but a practical precautionary measure.
Prompt for the KEY, or use the session value.

Uses the Session, so the key is not seen in the wild.

```suggestion
        const ably_key = sessionStorage.ABLY_KEY || prompt("Enter your key");
        if(/^[a-z0-9_\-.:]{32,64}$/i.test(ably_key)) sessionStorage.ABLY_KEY = ably_key;
```

amnonbc replied:
cool!
What happens if the key gets revoked?
Will the bad key be stuck in sessionStorage for ever?

fliptopbox replied:
umm yes and no. yes it is stuck (temporarily) and no, because it is the browser session (which is in memory), so it is forgotten when the browser restarts. the user will need to re-validate on first visit (per session).

amnonbc committed:
store ABLY_KEY in session storage

amnonbc committed:
remove stale comment

amnonbc committed:
store player colour in presence

Commit:
Recommended ES6 refactor

- also corrected base (fixes missing images issue)
- refactor main function to auto-execute in closure
- changed API prompt to insist until the key is legal

amnonbc committed:
Merge pull request #5 from ably-labs/add_js_viewer_bruce

Recommended ES6 refactor
